### PR TITLE
[Fix #1310] Fix a false positive for `RSpec/Capybara/SpecificMatcher`

### LINKED ISSUE #1310: [RSpec/Capybara/SpecificMatcher] False-positive with `have_table`
```rb
expect(page).to have_css('table tbody.some-class tr:last-child')
```

This results in a `Prefer have_table over have_css.` warning however in this case I have sub-matchers and that can't be replaced with `have_table`.

Otherwise the rule works perfect, great work!

cc @ydah

### PR DESCRIPTION
**Pull request by ydah**
Fix: #1310

Updated to not offense in the case of non-replaceable css selectors.

```ruby
expect(page).to have_css('button body')
expect(page).to have_css('a,h1')
expect(page).to have_css('table>tr')
expect(page).to have_css('select+option')
```

---

Before submitting the PR make sure the following are checked:

* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [-] Updated documentation.
* [x] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
* [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

If you have created a new cop:

* [-] Added the new cop to `config/default.yml`.
* [-] The cop is configured as `Enabled: pending` in `config/default.yml`.
* [-] The cop is configured as `Enabled: true` in `.rubocop.yml`.
* [-] The cop documents examples of good and bad code.
* [-] The tests assert both that bad code is reported and that good code is not reported.
* [-] Set `VersionAdded` in `default/config.yml` to the next minor version.

If you have modified an existing cop's configuration options:

* [-] Set `VersionChanged` in `config/default.yml` to the next major version.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 ## Master (Unreleased)
 
+* Fix a suggestions for safer conversions for `Lint/NonAtomicFileOperation`. ([@ydah][])
```
Comment (pirj) [inline CHANGELOG.md:5]: This seems to be from another PR.

**Review by pirj — APPROVED** [review]
Thank you for the quick fix!

File: CHANGELOG.md (hunk 1 above)
Comment (ydah) [reply CHANGELOG.md]: Oops, sorry. Fixed it.

**Review by bquorning — APPROVED** [review]
